### PR TITLE
Skip/speed up tests

### PR DESCRIPTION
**Pull request by rijobro**
Continued fix of https://github.com/Project-MONAI/MONAI/issues/1137.

### Description
Where possible we should speed slow tests up and if not possible, they should be given the `skip_if_quick` decorator.

### Status
**ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick`.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by rijobro** [comment]
Mostly made tensors smaller, added in missing `to(device)`, and if still too slow set to skip if quick. Quick tests now finish in <3m now.

Think the tests were mostly yours @wyli and @yiheng-wang-nv, so I'd be grateful if you could check that I haven't changed anything that wasn't desired.

**Review by wyli — APPROVED** [review]
thanks it looks good to me. @yiheng-wang-nv please confirm

File: tests/test_densenet.py [hunk 1]
```diff
@@ -60,100 +68,12 @@ def test_121_3d_shape_pretrain(self, input_param, input_shape, expected_shape):
...
-        net = densenet264(**input_param)
-        net.eval()
-        with torch.no_grad():
-            result = net.forward(torch.randn(input_shape))
-            self.assertEqual(result.shape, expected_shape)
-
-    @parameterized.expand([TEST_CASE_3])
-    def test_121_2d_shape(self, input_param, input_shape, expected_shape):
-        net = densenet121(**input_param)
+    @parameterized.expand([*TEST_CASES])
```
[the remark this answers is not in the document]
Comment (rijobro) [reply tests/test_densenet.py:71]: `TEST_CASES` is now a list of lists, so needs the first expansion.

[the remark this answers is not in the document]
Comment (wyli) [reply tests/test_densenet.py:71]: it's not an error, just wanted to say that `TEST_CASES == [*TEST_CASES]`, the syntax could be simplified a bit..

**Review by wyli — APPROVED** [review]

File: tests/test_densenet.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rijobro) [reply tests/test_densenet.py]: Oh, sure I see now. Thanks!

**Pull request merged**